Removed _compile workaround for create_block_mask

Chillee opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #137477

I also put in a change for supporting `create_block_mask` to properly handle non-multiples of BLOCK_SIZE.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

Chillee committed:
Removed _compile workaround for create_block_mask

[ghstack-poisoned]

Chillee committed:
Update on "Removed _compile workaround for create_block_mask"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang

[ghstack-poisoned]

Chillee committed:
Update on "Removed _compile workaround for create_block_mask"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang

[ghstack-poisoned]

Chillee committed:
Update on "Removed _compile workaround for create_block_mask"


I also put in a change for supporting `create_block_mask` to properly handle non-multiples of BLOCK_SIZE.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang

[ghstack-poisoned]

drisspg commented on torch/nn/attention/flex_attention.py:
```diff
@@ -883,18 +859,24 @@ def causal_mask(b, h, q_idx, kv_idx):
-    with TransformGetItemToIndex():
-        partial_block_mask, full_block_mask = inner_func(
-            mask_mod, B, H, Q_LEN, KV_LEN, device, Q_BLOCK_SIZE, KV_BLOCK_SIZE
+        warnings.warn(
```
Nit maybe a deprecation warning, it interacts much better with ides

drisspg reviewed (approved):
One comment, looks good

Chillee committed:
Update on "Removed _compile workaround for create_block_mask"


I also put in a change for supporting `create_block_mask` to properly handle non-multiples of BLOCK_SIZE.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang

[ghstack-poisoned]

Chillee committed:
Update on "Removed _compile workaround for create_block_mask"


I also put in a change for supporting `create_block_mask` to properly handle non-multiples of BLOCK_SIZE.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang

[ghstack-poisoned]

Chillee committed:
Update on "Removed _compile workaround for create_block_mask"


I also put in a change for supporting `create_block_mask` to properly handle non-multiples of BLOCK_SIZE.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang

[ghstack-poisoned]

Chillee committed:
Update on "Removed _compile workaround for create_block_mask"


I also put in a change for supporting `create_block_mask` to properly handle non-multiples of BLOCK_SIZE.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang

[ghstack-poisoned]

Chillee commented:
@pytorchbot merge -i
